kivy/uix/behaviors/knspace.py

```diff
@@ -0,0 +1,573 @@
+            self._knspace = value
+
+    knspace = AliasProperty(
+        _get_knspace, _set_knspace, bind=('_knspace', ), cache=False,
+        rebind=True, allownone=True)
+    '''The namespace instance, :class:`KNSpace`, associated with this widget.
+    The :attr:`knspace` namespace stores this widget when naming this widget
+    with :attr:`knsname`.
+
+    If the namespace has been set with a :class:`KNSpace` instance, e.g. with
+    `self.knspace = KNSpace()`, then that instance is returned (setting with
+    `None` doesn't count). Otherwise, if :attr:`knspace_key` is not None, we
+    look for a namespace to use in the object that is stored in the property
+    named :attr:`knspace_key`, of this instance. I.e.
+    `object = getattr(self, self.knspace_key)`.
+
+    If that object has a knspace property, then we return its value. Otherwise,
+    we go further up, e.g. with `getattr(object, self.knspace_key)` and look
+    for it's `knspace` property.
+
+    Finally, if we reach a value of `None`, or :attr:`knspace_key` was `None`,
+    the default :attr:`~kivy.uix.behaviors.knspace.knspace` namespace is
+    returned.
+
+    If :attr:`knspace` is set to the string `'fork'`, the current namespace
+    in :attr:`knspace` will be forked with :meth:`KNSpace.fork` and the
+    resulting namespace will be assigned to this instance's :attr:`knspace`.
+    See the module examples for a motivating example.
+
+    Both `rebind` and `allownone` are `True`.
+    '''
+
+    knspace_key = StringProperty('parent', allownone=True)
+    '''The name of the property of this instance, to use to search upwards for
+    a namespace to use by this instance. Defaults to `'parent'` so that we'll
+    search the parent tree. See :attr:`knspace`.
+
+    When `None`, we won't search the parent tree for the namespace.
+    `allownone` is `True`.
+    '''
+
+    def _get_knsname(self):
+        return self._knsname
+
+    def _set_knsname(self, value):
+        old_name = self._knsname
+        knspace = self.knspace
+        if old_name and knspace and getattr(knspace, old_name) == self:
+            setattr(knspace, old_name, None)
+
+        self._knsname = value
+        if value:
+            if knspace:
+                setattr(knspace, value, self)
+            else:
+                raise ValueError('Object has name "{}", but no namespace'.
+                                 format(value))
+
+    knsname = AliasProperty(
+        _get_knsname, _set_knsname, bind=('_knsname', ), cache=False)
+    '''The name given to this instance. If named, the name will be added to the
+    associated :attr:`knspace` namespace, which will then point to the
+    `proxy_ref` of this instance.
+
+    When named, one can access this object by e.g. self.knspace.name, where
+    `name` is the given name of this instance. See :attr:`knspace` and the
+    module description for more details.
+    '''
+
+knspace = KNSpace()
```

Comment:
You missed the documentation for that:

    #: default instance of :class:`KNSpace`